### PR TITLE
Allow a board to be built and asserted using ASCII

### PR DESCRIPTION
**Pull request by DavidBetteridge**
You set up PacMan and the ghosts as before.  When you call CreateBoard it uses PacMan to position the rest of the pieces,  and if it sees a G it checks you have put a ghost there.

AssertBoard uses the position of PacMan it worked out during CreateBoard to assert that all the pieces are in the correct location.  Again if it sees a G it checks there is a ghost in that square.  If you have 2 ghosts in the same location then it would be marked as a single G on the map.

```
   [Fact]
        public async Task GhostShouldBeInMiddleOfHouseAfterBeingEatingAndWaitingForElapsedTime()
        {
            var ghostStart1 = _gameSettings.PacMan.Location.Below.Below.Left.Left;
            var ghost1 = GhostBuilder.New()
                                     .WithLocation(ghostStart1)
                                     .WithChaseStrategyRight()
                                     .Create();
            _gameSettings.Ghosts.Add(ghost1);
            _gameSettings.CreateBoard("    P              ",
                                      "  . *   # # - - # #",
                                      "G . .   # H H H H #",
                                      "        # H H H H #",
                                      "        # H H H H #",
                                      "        # # # # # #");
            var gameHarness = new GameHarness(_gameSettings);
            gameHarness.Game.StartGame();
            await gameHarness.ChangeDirection(Direction.Down);
            await gameHarness.EatPill();
            gameHarness.AssertBoard("                   ",
                                    "  . P   # # - - # #",
                                    "  G .   # H H H H #",
                                    "        # H H H H #",
                                    "        # H H H H #",
                                    "        # # # # # #");
            await gameHarness.EatGhost(ghost1);
            gameHarness.AssertBoard("                   ",
                                    "  .     # # - - # #",
                                    "    P   # H H H H #",
                                    "        # H H H H #",
                                    "        # H H H H #",
                                    "        # # # # # #");
```

### CODE REVIEW

File: test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            foreach (var row in rows)
+            {
+                AssertRow(game, row, currentCell);
+                currentCell = currentCell.Below;
+            }
+        }
+
+        private void RemoveCurrentSettings()
+        {
+            _gameSettings.GhostHouse.Clear();
```
Comment (DavidBetteridge) [inline test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs:46]: Should we also remove coins etc

File: test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+            var currentCell = _topLeft;
+            foreach (var row in rows)
+            {
+                CreateRow(row, currentCell);
+                currentCell = currentCell.Below;
+            }
+        }
+
+        internal void AssertBoard(Game game, params string[] rows)
```
Comment (DavidBetteridge) [inline test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs:32]: We should check that CreateBoard has been called

File: test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    class AsciiTestSettingBuilder
+    {
+        private readonly TestGameSettings _gameSettings;
+        private CellLocation _topLeft;
+
+        public AsciiTestSettingBuilder(TestGameSettings gameSettings)
+        {
+            _gameSettings = gameSettings;
+        }
+        internal void CreateBoard(params string[] rows)
```
Comment (DavidBetteridge) [inline test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs:16]: We should check that CreateBoard hasn't already been called

File: test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                    if (cell == 'P')
+                        return new CellLocation(x, y);
+
+                    x++;
+                }
+
+                y++;
+            }
+
+            throw new Exception("Could not find pac man");
```
Comment (DavidBetteridge) [inline test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs:89]: Should also report if we find multiple PacMan

File: test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs (hunk 1 above)
Comment (kevbite) [reply test/NPacMan.Game.Tests/AsciiTestSettingBuilder.cs:46]: Would be a bit iffy as we've got go put a coin FarAway from where we're playing so we don't complete the level.

**Pull request merged**